src/sentry/models/grouphashmetadata.py

```diff
@@ -56,6 +58,12 @@ class GroupHashMetadata(Model):
     latest_grouping_config = models.CharField(null=True)
     # The primary grouping method (message, stacktrace, fingerprint, etc.)
     hash_basis = models.CharField(choices=HashBasis, null=True)
+    # Metadata about the inputs to the hashing process and the hashing process itself (what
+    # fingerprinting rules were matched? did we parameterize the message? etc.). For the specific
+    # data stored, see the class definitions of the `HashingMetadata` subtypes.
+    hashing_metadata: models.Field[HashingMetadata | None, HashingMetadata | None] = JSONField(
```

Comment:
why does this need to be repeated `HashingMetadata | None, HashingMetadata | None`?

Reply:
Not well documented, but this is the setter type and the getter type: https://github.com/typeddjango/django-stubs/blob/ab4dcfa002dbe138b7548086328f13cbde3c41b3/django-stubs/db/models/fields/__init__.pyi#L46-L51.